runsc/fsgofer/fsgofer.go

```diff
@@ -124,30 +130,54 @@ func (a *attachPoint) Attach() (p9.File, error) {
 	if err != nil {
 		return nil, fmt.Errorf("stat file %q, err: %v", a.prefix, err)
 	}
-	mode := syscall.O_RDWR
-	if a.conf.ROMount || (stat.Mode&syscall.S_IFMT) == syscall.S_IFDIR {
-		mode = syscall.O_RDONLY
-	}
-
-	// Open the root directory.
-	f, err := fd.Open(a.prefix, openFlags|mode, 0)
-	if err != nil {
-		return nil, fmt.Errorf("unable to open file %q, err: %v", a.prefix, err)
-	}
 
+	// Acquire the attach point lock
 	a.attachedMu.Lock()
 	defer a.attachedMu.Unlock()
+
+	// Hold the file descriptor we are converting into a p9.File
+	var f *fd.FD
+
+	// Apply the S_IFMT bitmask so we can detect file type appropriately
+	switch fmtStat := stat.Mode & syscall.S_IFMT; {
+	case fmtStat == syscall.S_IFSOCK:
+		// Check to see if the CLI option has been set to allow the UDS mount
```

Comment:
Please add a period at the end of a sentence. Here and the other new comments as well.